File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/http/HttpServer.php
<?php 
namespace io\vertx\php\core\http;
use io\vertx\php\core\streams\ReadStream;
class HttpServer
{
    
    private function __construct()
    {

    }
    /**
     * ()
     * @return int
     * methodSize : 1
     */
    public function actualPort(/*0*/)
    {

    }
    /**
     * ()
     * @return Void
     * (callable completionHandler)
     * @return Void
     * methodSize : 2
     */
    public function close(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return HttpServer
     * methodSize : 1
     */
    public function connectionHandler(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return HttpServer
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isMetricsEnabled(/*0*/)
    {

    }
    /**
     * ()
     * @return HttpServer
     * (int port, string host)
     * @return HttpServer
     * (int port, callable listenHandler)
     * @return HttpServer
     * (int port, string host, callable listenHandler)
     * @return HttpServer
     * (int port)
     * @return HttpServer
     * (callable listenHandler)
     * @return HttpServer
     * methodSize : 6
     */
    public function listen(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * (callable handler)
     * @return HttpServer
     * methodSize : 1
     */
    public function requestHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return ReadStream
     * methodSize : 1
     */
    public function requestStream(/*0*/)
    {

    }
    /**
     * (callable handler)
     * @return HttpServer
     * methodSize : 1
     */
    public function websocketHandler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return ReadStream
     * methodSize : 1
     */
    public function websocketStream(/*0*/)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/http/HttpServer.java
package io.vertx.php.core.http;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.http.HttpServerRequest;
import io.vertx.php.core.http.ServerWebSocket;
import io.vertx.php.core.metrics.Measured;
import io.vertx.php.core.streams.ReadStream;
import io.vertx.php.core.http.HttpConnection;
@Name("HttpServer")
@Namespace("io\\vertx\\php\\core\\http")
public interface HttpServer<S extends io.vertx.core.http.HttpServer> extends IWrapper<S>{
    
    @Signature
    default Memory isMetricsEnabled(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isMetricsEnabled()
    }
    
    @Signature
    default Memory requestStream(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, ReadStreamImpl::new, this.getWrappedObject().requestStream()
    }

    @Signature
    default Memory requestHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().requestHandler(handler)
    }

    @Signature
    default Memory connectionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().connectionHandler(handler)
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }
    
    @Signature
    default Memory websocketStream(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, ReadStreamImpl::new, this.getWrappedObject().websocketStream()
    }

    @Signature
    default Memory websocketHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().websocketHandler(handler)
    }
    
    @Signature
    default Memory listen(Environment __env__, Memory... args) {
        this.getWrappedObject().listen()
    }

    @Signature
    default Memory listen(Environment __env__, Memory... args) {
        this.getWrappedObject().listen()
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }

    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }
    
    @Signature
    default Memory actualPort(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().actualPort()
    }

}
